### PR TITLE
Add musdb18 recipe for music source separation

### PR DESCRIPTION
**Pull request by Emrys365**
This PR adds a new recipe for music source separation.

I followed the [musdb package](https://github.com/sigsep/sigsep-mus-db/blob/master/musdb/configs/mus.yaml) to obtain the training-validation splits.

### CODE REVIEW

File: egs2/musdb18/enh1/local/data.sh [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+fi
+
+if [ -z "${MUSDB18}" ]; then
+    log "Fill the value of 'MUSDB18' of db.sh"
+    exit 1
+fi
+
+if [ $nchannels -eq 1 ]; then
+    log "Use single channel audios"
+elif [ $nchannels -eq 1 ]; then
```
Comment (YoshikiMas) [inline egs2/musdb18/enh1/local/data.sh:65]: It should be `-eq 2`, right?

Comment (Emrys365) [reply egs2/musdb18/enh1/local/data.sh]: Oh thanks for noticing that! I have fixed it.

File: egs2/musdb18/enh1/local/data.sh [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+            sed -i -e 's/^\(.*\)/\1_CH1\n\1_CH2/g' "${cdir}/data/${dset}/uid"
+        fi
+        paste -d' ' "${cdir}/data/${dset}/uid" "${cdir}/data/${dset}/flist" | sort -u > "${cdir}/data/${dset}/wav.scp"
+        awk '{print($1 " " $1)}' "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/utt2spk"
+        utils/utt2spk_to_spk2utt.pl "${cdir}/data/${dset}/utt2spk" > "${cdir}/data/${dset}/spk2utt"
+
+        sed -e "s#/mixture.wav#/drums.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk1.scp"
+        sed -e "s#/mixture.wav#/bass.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk2.scp"
+        sed -e "s#/mixture.wav#/vocals.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk3.scp"
+        sed -e "s#/mixture.wav#/other.wav#g" "${cdir}/data/${dset}/wav.scp" > "${cdir}/data/${dset}/spk4.scp"
```
Comment (ftshijt) [inline egs2/musdb18/enh1/local/data.sh:154]: Is it possible to add a readme in this recipe, where we can discuss the mapping

**Review by ftshijt — COMMENTED** [review]
LGTM! A minor suggestion as follows:

File: egs2/musdb18/enh1/local/data.sh (hunk 2 above)
Comment (sw005320) [reply egs2/musdb18/enh1/local/data.sh:154]: @Emrys365, please do so in your follow up PR.

Comment (Emrys365) [reply egs2/musdb18/enh1/local/data.sh:154]: Sure.

**Pull request merged**